pkg/cloud/services/ec2/instances.go

```diff
@@ -130,6 +131,13 @@ func (s *Service) CreateInstance(scope *scope.MachineScope) (*infrav1.Instance,
 	if scope.AWSMachine.Spec.AMI.ID != nil {
 		input.ImageID = *scope.AWSMachine.Spec.AMI.ID
 	} else {
+		if scope.Machine.Spec.Version == nil {
+			err := errors.New("Either AWSMachine's spec.ami.id or Machine's spec.version must be defined")
+			scope.SetFailureReason(capierrors.CreateMachineError)
+			scope.SetFailureMessage(err)
+			return nil, err
```

Comment:
It doesn't look like returning an error here will generate an event. Errors returned from `CreateInstance` here are wrapped and returned in the MachineControllers `getOrCreate` method, which are then returned from the MachineControllers `reconcileNormal` method, which is directly returned from the primary `Reconcile` method.

Reply:
We don't strictly need an event because it will be visible in status.failure*

Reply:
probably would benefit from an event. I'll add one unless there is an objection

Reply:
Already merged, but let's not 😄

Reply:
I figured we can take the conversation asynchronous :joy: